Clearer default error messages

varjolintu opened the pull request:
[TIP]:  # ( Provide a general summary of your changes in the title above ^^ )

Modify the error messages to be a little more clearer. Any suggestions are very welcome.

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Bug fix (non-breaking change which fixes an issue)

## Description and Context
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Describe the context of your change. Explain large code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" as necessary )
The previous error messages are not clear enough to describe the error situation.

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )
Manually.

## Checklist:
[NOTE]: # ( Please go over all the following points. )
[NOTE]: # ( Again, remove any lines which don't apply. )
[NOTE]: # ( Pull Requests that don't fulfill all [REQUIRED] requisites are likely )
[NOTE]: # ( to be sent back to you for correction or will be rejected.  )
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**

droidmonkey commented on src/browser/BrowserAction.cpp:
```diff
@@ -406,11 +406,11 @@ QString BrowserAction::getErrorMessage(const int errorCode) const
         return QObject::tr("Cannot decrypt message");
     case ERROR_KEEPASS_TIMEOUT_OR_NOT_CONNECTED:
-        return QObject::tr("Timeout or cannot connect to KeePassXC");
+        return QObject::tr("Cannot connect to KeePassXC. Is Browser Integration enabled?");
```
Recommend second sentence of: "Check that browser integration is enabled in KeePassXC settings."

droidmonkey commented on src/browser/BrowserAction.cpp:
```diff
@@ -406,11 +406,11 @@ QString BrowserAction::getErrorMessage(const int errorCode) const
         return QObject::tr("Action cancelled or denied");
     case ERROR_KEEPASS_CANNOT_ENCRYPT_MESSAGE:
-        return QObject::tr("Cannot encrypt message or public key not found. Is Native Messaging enabled in KeePassXC?");
+        return QObject::tr("Message encryption failed. Is KeePassXC open and connected?");
```
"Is KeePassXC running?" I don't think there is an obvious way to see if it is "connected" otherwise this error message wouldn't show up.

droidmonkey commented on src/browser/BrowserAction.cpp:
```diff
@@ -406,11 +406,11 @@ QString BrowserAction::getErrorMessage(const int errorCode) const
         return QObject::tr("Action cancelled or denied");
     case ERROR_KEEPASS_CANNOT_ENCRYPT_MESSAGE:
-        return QObject::tr("Cannot encrypt message or public key not found. Is Native Messaging enabled in KeePassXC?");
+        return QObject::tr("Message encryption failed. Is KeePassXC running?");
```
Hmmm, I was confused when I first reviewed this PR. Of course KeePassXC is running or you wouldn't get this error message in the first place. Perhaps just leave it at "Message encryption failed."

droidmonkey replied:
This error code is not used anywhere in the code.

droidmonkey commented on src/browser/BrowserAction.cpp:
```diff
@@ -406,11 +406,11 @@ QString BrowserAction::getErrorMessage(const int errorCode) const
         return QObject::tr("Cannot decrypt message");
     case ERROR_KEEPASS_TIMEOUT_OR_NOT_CONNECTED:
-        return QObject::tr("Timeout or cannot connect to KeePassXC");
+        return QObject::tr("Cannot connect to KeePassXC. Check that browser integration is enabled in KeePassXC settings.");
```
This doesn't make sense. If I am timed out or not connected how the heck is KeePassXC sending this error message to the extension??

droidmonkey replied:
Also, this error code is not used anywhere in the code. ERROR_KEEPASS_TIMEOUT_OR_NOT_CONNECTED

droidmonkey commented:
Other error codes not used are:

ERROR_KEEPASS_KEY_CHANGE_FAILED
ERROR_KEEPASS_ENCRYPTION_KEY_UNRECOGNIZED
ERROR_KEEPASS_NO_SAVED_DATABASES_FOUND
ERROR_KEEPASS_EMPTY_MESSAGE_RECEIVED

varjolintu commented:
@droidmonkey Those unused error messages are now removed. The reason for those was to keep the similarity with the browser extension, but that's not really needed. I also added some error messages to certain points that could be useful. Thanks for the review.
